@here/harp-lines/lib/TriangulateLines.ts

```diff
@@ -95,59 +95,60 @@ export function triangulateLine(
 
     const baseVertex = vertices.length / 3;
 
-    const prevDir = new THREE.Vector3();
-    const p = new THREE.Vector3();
-    const n = new THREE.Vector3();
-    const dir = new THREE.Vector3();
-    const aver = new THREE.Vector3();
+    // bt = Bitangent (i.e. extrusion vector)
+    const prevBt = new THREE.Vector3();
+    const p = new THREE.Vector3(); // current point
+    const n = new THREE.Vector3(); // next point
+    const bt = new THREE.Vector3();
+    const averageBt = new THREE.Vector3();
     const p0 = new THREE.Vector3();
     const p1 = new THREE.Vector3();
     const p2 = new THREE.Vector3();
     const p3 = new THREE.Vector3();
 
-    const N = points.length / 2;
+    const N = points.length / 3;
```

Comment:
That was the actual bug. All other changes are just renaming to make the code more readable.